building-protected-and-paywall.mdx

```diff
@@ -0,0 +1,969 @@
+
+```tsx
+// File: src/pages/api/sign/out.ts
+
+import { lucia } from '@/lucia/index';
+import type { APIContext } from 'astro';
+import { getSessionID } from '@/lucia/user';
+
+export async function GET({ cookies, redirect }: APIContext): Promise<Response> {
+  await lucia.invalidateSession(getSessionID(cookies));
+  const sessionCookie = lucia.createBlankSessionCookie();
+  cookies.set(sessionCookie.name, sessionCookie.value, sessionCookie.attributes);
+  return redirect('/');
+}
+```
+
+The code above does the following:
+
+- Imports the initialized `lucia` instance and `getSessionID` helper utility.
+- Exports a GET HTTP handler to only respond to incoming GET request(s).
+- Invalidates the current user session using the `invalidateSession` utility by Lucia.
+- Creates an empty user session cookie using `createBlankSessionCookie` utility by Lucia.
+- Sets the blank user session cookie.
+- Redirects to the index route.
+
+#### Create Protected and Paid Routes
+
+To restrict access to a page for only paid and authenticated users, you will obtain the user information using the utilities created earlier. Create a file `src/pages/protected_and_paid.astro` with the following code:
+
+```html
+---
+// File: src/pages/protected_and_paid.astro
+
+import { getUser } from '@/lucia/user'
+
+const user = await getUser(Astro.cookies)
+
+if (!user || user.paid !== true) return new Response(null, { status: 403 })
+---
+
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width" />
+  </head>
+  <body class="font-display overflow-x-hidden">
+    <span> Protected and Paid Content </span>
+  </body>
+</html>
+```
+
+The code above creates an `/protected_and_paid` route in the Astro application that does the following:
+
+- Checks if there's a user associated with the session and if they have paid.
+- If the above is not true, it returns a 403 Unauthorized response. Otherwise, it returns a HTML page containing the paid and protected information.
+
+To restrict access to a page for only authenticated users, you will obtain the user information using the utilities created earlier. Create a file `src/pages/protected.astro` with the following code:
+
+```html
+---
+import { getUser } from '@/lucia/user'
+
+const user = await getUser(Astro.cookies)
+
+if (!user) return new Response(null, { status: 403 })
+---
+
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <meta name="viewport" content="width=device-width" />
+  </head>
+  <body class="font-display overflow-x-hidden">
+    <span> Protected Content </span>
+  </body>
+</html>
+```
+
+The code above creates an `/protected` route in the Astro application that does the following:
+
+- Checks if there's a user associated with the session.
+- If the above is not true, it returns a 403 Unauthorized response. Otherwise, it returns a HTML page containing the protected information.
```

Comment:
```suggestion
- If the above is not true, it returns a 403 Unauthorized response. Otherwise, it returns an HTML page containing the protected information.
```